src/test/typescript/typescript-analyzer_test.ts

```diff
@@ -0,0 +1,105 @@
+/**
+ * @license
+ * Copyright (c) 2017 The Polymer Project Authors. All rights reserved.
+ * This code may only be used under the BSD style license found at
+ * http://polymer.github.io/LICENSE.txt
+ * The complete set of authors may be found at
+ * http://polymer.github.io/AUTHORS.txt
+ * The complete set of contributors may be found at
+ * http://polymer.github.io/CONTRIBUTORS.txt
+ * Code distributed by Google as part of the polymer project is also
+ * subject to an additional IP rights grant found at
+ * http://polymer.github.io/PATENTS.txt
+ */
+
+import {assert} from 'chai';
+import * as ts from 'typescript';
+
+import {AnalysisContext} from '../../core/analysis-context';
+import {TypeScriptAnalyzer} from '../../typescript/typescript-analyzer';
+import {UrlLoader} from '../../url-loader/url-loader';
+import {UrlResolver} from '../../url-loader/url-resolver';
+
+class TestUrlResolver implements UrlResolver {
```

Comment:
I really like being able to provide small test cases as strings. I'm inclined to add an `InMemoryUrlLoader` that takes a map of files to make this even easier.